scripts/build.js

```diff
@@ -327,6 +352,32 @@ async function copyComponentsJson () {
 	await copyFile(from, to);
 }
 
+/**
+ * Recursively find all files with a given extension in a directory.
+ *
+ * @param {string} dir
+ * @param {string} [ext='.js']
+ * @returns {Promise<string[]>}
+ */
+async function findFiles (dir, ext = '.js') {
+	const files = [];
+	const items = await readdir(dir);
+
+	for (const item of items) {
+		const fullPath = path.join(dir, item);
+		const stats = await stat(fullPath);
+
+		if (stats.isDirectory()) {
+			files.push(...(await findFiles(fullPath)));
+		}
+		else if (path.extname(item) === ext) {
+			files.push(fullPath);
+		}
+	}
+
+	return files;
+}
```

Comment:
I adjusted the process of building types, so it would be nice if you could review this part.